### PR TITLE
new package: espeak-ng-1.50

### PR DESCRIPTION
**Pull request by joey-commits**
Speech Dispatcher and espeakup were also corrected to build against the new espeak-ng
  I used dummy packages to replace Espeak on installed machines. The idea was gained from polkit-elogind, however, I am open to doing something else if it is prefered.  Apparently, the author of espeak could not be contacted and espeak-ng was created a few years ago to continue development.
I have been using these packages with my screen readers for a few hours and everything seems like its working so far.

### CODE REVIEW

**Comment by joey-commits** [comment]
I am not sure how to fix the builds on non-intel arches. I guess its failing because its trying to execute the non-intel binaries on the intel builder as part of the build process. I will do some research and would also appreciate any help if someone has time. I'll try to figure this out but cross compiling is new territory for me.

**Comment by ericonr** [comment]
Please fix the xlint complaints:

```
Run common/travis/xlint.sh
Linting srcpkgs/espeak-ng/template...
srcpkgs/espeak-ng/template: Place checksum= after homepage=
Linting srcpkgs/espeakup/template...
srcpkgs/espeakup/template: Place checksum= after build_style=
srcpkgs/espeakup/template: Place distfiles= after makedepends=
srcpkgs/espeakup/template:5: short_desc should not start with an article
srcpkgs/espeakup/template:7: use SPDX id for 'GPL-3' license or see Manual.md
Linting srcpkgs/speech-dispatcher/template...
Error: Process completed with exit code 1.
```

**Comment by ericonr** [comment]
Argh, solving this might be a bit complicated, but definitely required, since espeak did cross build previously. I will try to poke at it.

**Comment by joey-commits** [comment]
I was looking at the data built by the binaries and it doesn't seem to be archetecture specific. I need to look at a bit more to be sure. I am wondering if I  could make a  nocross package to build the data files and then pull it in for the real espeak-ng package with hostmakedepends. Then it could be coppied to the build directory before the build and the Makefile would detect that the voice dictionaries are already built.

**Comment by ericonr** [comment]
> I am wondering if I could make a nocross package to build the data files and then pull it in for the real espeak-ng package with hostmakedepends. Then it could be coppied to the build directory before the build and the Makefile would detect that the voice dictionaries are already built.

Doesn't even need to be nocross. I've done exactly this for the `sdcc` package. Something like:

```
if [ "$CROSS_BUILD" ]; then
hostmakedepends+=" espeak-ng"
fi

pre_build() {
if [ "$CROSS_BUILD" ]; then
cp -r ....
fi
}
```

File: srcpkgs/espeakup/template [hunk 1]
```diff
@@ -1,15 +1,16 @@
 # Template file for 'espeakup'
 pkgname=espeakup
 version=0.80
-revision=1
-short_desc="A lightweight connector for espeak and speakup"
+revision=2
+short_desc="Lightweight connector for espeak and speakup"
 maintainer="Rui Abreu Ferreira <raf-ep@gmx.com>"
-license="GPL-3"
+
```
Comment (Piraty) [inline srcpkgs/espeakup/template:7]: please xlint this template

**Review by Piraty — CHANGES_REQUESTED** [review]

**Comment by Piraty** [comment]
if you'd be so kind to ensure arcan's tts build option will work with your changeset

File: srcpkgs/espeak-ng/template [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+pre_configure() {
+	./autogen.sh
+}
+
+post_configure() {
+	if [ "$CROSS_BUILD" ]; then
+		sed -i '2785,2818 s_src/espeak-ng_/usr/bin/espeak-ng_' Makefile
+		sed -i '2819,$ s_../src/espeak-ng_/usr/bin/espeak-ng_' Makefile
+				sed -i '3244,$ s_src/espeak-ng_/usr/bin/espeak-ng_' Makefile
+	fi
```
Comment (ericonr) [inline srcpkgs/espeak-ng/template:29]: I guess just putting the files there didn't work? This is quite unfortunate.

Having numbered seds is quite fragile, though, so ideally we could come up with a patch for it.

Also, indentation of the last `sed` line is too far right by 2 tabs.

**Comment by joey-commits** [comment]
I am just glad its building. Gives me hope. Sorry about the indentation issues. I am trying to get better about this, however, indentation is tricky when using a screen reader. It makes me wonder how many other people are actually packaging and programming with Orca instead of a screen. I don't know anyone else.

**Comment by joey-commits** [comment]
I can definitely do a patch instead. Are you ok with me patching the Makefile after running configure like I am doing with sed?

**Comment by ericonr** [comment]
>  I am trying to get better about this, however, indentation is tricky when using a screen reader.

I can imagine. If you need any help for the fiinal clean up, I can do a final once over. Thanks for contributing :)

> I can definitely do a patch instead. Are you ok with me patching the Makefile after running configure like I am doing with sed?

Hmm that is indeed not ideal, since it can change dependending on the environment. Can you show me the part you're trying to delete? We can try to come up with a solution.

**Comment by joey-commits** [comment]
The main problem is that the Makefile calls espeak-ng in a bunch of places. I am simply redirecting it to usr/bin in all those cases so that it runs the copy from the x86_64 package. I am really sorry. i don't know enough about autotools to fix this properly.

**Comment by joey-commits** [comment]
After espeak-ng is compiled, the binary is used to compile the dictionaries and phoneme data as a part of the make process..

**Comment by ericonr** [comment]
Ah! Got it.

I think if we can make the rules a bit more specific, it won't be necessary to specify the line numbers. I will try to build it locally to see if I can come up with one, but I think `vsed -e "s_\ts_src/espeak-ng_/usr/bin/espeak-ng_" -i Makefile` might do the trick (plus a rule with `../s_src/espeak-ng`.

By the way, do you think we could use a symlink in `/usr/bin/espeak`?

**Comment by joey-commits** [comment]
I think a simlink pointing from /usr/bin/espeak to /usr/bin/espeak-ng could be nice. I will add that to the espeak-ng package. Thanks so much for your help with this. Espeak-ng works so much better with my reader and it will be nice to have it in the repositories.

**Comment by joey-commits** [comment]
@piraty Is there any way you could help me test Arcan? It looks like an alternative display server. I think my screen reader depends on Xorg and may not run in Arcan. From what I can tell, it ships with its own desktop and I think Orca only supports Gnome, Mate and XFCE.

**Comment by joey-commits** [comment]
A quick grep of the Arcan sources shows that it uses a header that espeak-ng ships for backwards compatibility. I would think it would work ok.

**Comment by Piraty** [comment]
for now it's enough if arcan builds and correctly links the espeak-ng, as it's just an opt-in build option it's for custom use cases anyway. thanks for you work!

**Comment by joey-commits** [comment]
Hey, I was wondering if there are any further changes you think we would need before this can enter the repo. I think the tests failed du to pulseaudio 14 tests. I replaced the line numbers with searches and substitutions like we discussed. Thanks again for all the help!

**Comment by joey-commits** [comment]
I mean the packages failed to build due to pulseaudio tests failing not my packages not building.

**Comment by ericonr** [comment]
I'm re-running all CI jobs, then. How can I test the package?

**Comment by joey-commits** [comment]
You can test espeak-ng by typing:

espeak-ng "testing testing"

 which will play speech through your sound card.

Alternatively:

espeak-ng "testing testing" -w output.wav

will generate a PCM audio file with the speech.

**Comment by joey-commits** [comment]
Looks like this is passing since pulse 14 hit the repos. That is good news.

**Comment by ericonr** [comment]
@joey-commits 

Another espeak user (over IRC) asked that we keep both espeak and espeak-ng as options, since they had heard that espeak-ng had some issues with latency. What are your thoughts on that? If you think it's ok but don't want to work on the template, I can try to get it going. I believe it would be possible to have both available, including as a build option for speech-dispatcher.

**Comment by ericonr** [comment]
I'd imagine it doesn't integrate nearly as well, but have you tried `mimic`? It sounds reasonably good.

**Comment by joey-commits** [comment]
I would be ok with keeping espeak but I would be very interested to learn the use case. Since espeak is not maintained any more, I would much   rather keep espeak around until we can fix all issues and then migrate. I certainly wouldn't want to break someones workflow.
I know that speech dispatcher has been less latent with espeak-ng for a long time. I also applied patches to pcaudilib to ensure that espeak-ng would be less latent with espeakup. The patches that I applied fix two problems with espeakup which are latency and overlapping audio. I tested espeakup for a while today to ensure that it was reliable since I actually have friends who use it even though I am mostly an Orca user.

**Comment by joey-commits** [comment]
The problem with keeping both around is that you can only build espeakup against one or the other. I have been thinking about wanting to migrate to espeak-ng for a long time but I knew about the espeakup issues. I didn't want to make a pull request until I was sure those issues were solved. One of the accessibility folks who works on the debian project solved the accessibility issues with espeakup.. If there is another thing though that is broken that I don't know about, lets the use case and wait to migrate until we can be sure it is solved. Having both around will be messy.

**Comment by joey-commits** [comment]
Could the user comment on the pull request here? I think it would be good to keep a record of the issues people have on this pull request so that we can think about and solve them.

**Comment by ericonr** [comment]
> One of the accessibility folks who works on the debian project solved the accessibility issues with espeakup..

That's nice to hear!

What I was considering was adding a new espeak-ng package which doesn't replace espeak, and adding a build option to espeakup to allow it to be built with either, but make espeak-ng the default. The only disadvantage is that an espeak user who hasn't heard about espeak-ng won't get an update.

I asked them to comment, but haven't heard back yet.

**Comment by CMB** [comment]
I've been testing this PR.  There are no latency issues anymore.  I
whole-heartedly approve.

**Comment by CMB** [comment]
BTW this is teiresias from freenode.

**Comment by ericonr** [comment]
Yay! Thank you very much for checking :)

**Review by Piraty — APPROVED** [review]

**Pull request closed without merging**